src/hooks/useNotifications.ts

```diff
@@ -312,6 +317,47 @@ export const useNotifications = (colors: boolean): NotificationsState => {
     [notifications],
   );
 
+  const markRepoNotificationsDone = useCallback(
+    async (accounts, repoSlug, hostname) => {
+      setIsFetching(true);
+
+      try {
+        const accountIndex = notifications.findIndex(
+          (accountNotifications) => accountNotifications.hostname === hostname,
+        );
+
+        if (accountIndex !== -1) {
+          const notificationsToRemove = notifications[
+            accountIndex
+          ].notifications.filter(
+            (notification) => notification.repository.full_name === repoSlug,
+          );
+
+          for (const notification of notificationsToRemove) {
+            await markNotificationDone(
```

Comment:
Is it possible/worth it to run these in parallel?

I guess we'd probably want to batch them, and then that blows up in scope pretty quickly, so maybe not?

Just asking questions :)

Reply:
That's a good point, I think it's better to run them in parallel, I've updated my code to use a [await Promise.all()](https://github.com/adufr/gitify/blob/bb609b9b829494e9a1b15ecd315a541d74d1d960/src/hooks/useNotifications.ts#L336-L344) 🙂

It also helps with the animation when notifications are removed because they were removed in a weird order, now they all disappear at the same time